mmv1/products/bigqueryconnection/Connection.yaml

```diff
@@ -299,18 +299,48 @@ properties:
       - !ruby/object:Api::Type::String
         name: 'database'
         description:
-          Cloud Spanner database in the form `project/instance/database'
+          Cloud Spanner database in the form `project/instance/database'.
         required: true
       - !ruby/object:Api::Type::Boolean
         name: 'useParallelism'
         description:
-          If parallelism should be used when reading from Cloud Spanner
+          If parallelism should be used when reading from Cloud Spanner.
+      - !ruby/object:Api::Type::Integer
+        name: 'maxParallelism'
+        description:
+          Allows setting max parallelism per query when executing on Spanner independent compute
+          resources. If unspecified, default values of parallelism are chosen that are dependent on
+          the Cloud Spanner instance configuration. `useParallelism` and `useDataBoost` must be set
+          when setting max parallelism.
+        required_with:
+          - useDataBoost
+          - useParallelism
+      - !ruby/object:Api::Type::Boolean
+        name: 'useDataBoost'
+        description:
+          If set, the request will be executed via Spanner independent compute resources.
+          `use_parallelism` must be set when using data boost.
+        required_with:
+          - useParallelism
+      - !ruby/object:Api::Type::String
+        name: 'databaseRole'
+        description:
+          Cloud Spanner database role for fine-grained access control. The Cloud Spanner admin
+          should have provisioned the database role with appropriate permissions, such as `SELECT`
+          and `INSERT`. Other users should only use roles provided by their Cloud Spanner admins.
+          The database role name must start with a letter, and can only contain letters, numbers,
+          and underscores. For more details, see https://cloud.google.com/spanner/docs/fgac-about.
+        validation: !ruby/object:Provider::Terraform::Validation
+          regex: '^[a-zA-Z][a-zA-Z0-9_]*$'
       - !ruby/object:Api::Type::Boolean
         name: 'useServerlessAnalytics'
         description:
           If the serverless analytics service should be used to read data from
-          Cloud Spanner. useParallelism must be set when using serverless
-          analytics
+          Cloud Spanner. `useParallelism` must be set when using serverless
+          analytics.
+        deprecation_message: >-
+          `useServerlessAnalytics` is deprecated and will be removed in a future major release. Use
+          `useDataBoost` instead.
```

Comment:
Talking with my team on this.

This line doesn't appear to be true
`useParallelism` must be set when using serverless analytics as there is existing. As `useParallelism` didn't exist when `useServerlessAnalytics` was supported.


Question, can you add a test showing that you can update from `useServerlessAnalytics` to `useDataBoost` within an update. Through a handwritten test perhaps? This is needed to allow a deprecation message within a minor release.

Reply:
As it stands today the server will throw `INVALID_INPUT` if only `userServerlessAnalytics` is specified alone. Not sure whether this has always been the case or only recent. It'll be a breaking change to add `required_with` to it so I'm holding off especially because it's being deprecated.

Added `TestAccBigqueryConnectionConnection_bigqueryConnectionCloudSpannerServerlessAnalyticsToDataBoostUpdate`.